MdeModulePkg/Library/UefiHiiLib/HiiLib.c

```diff
@@ -2131,11 +2158,22 @@ GetBlockDataInfo (
   while ((Link != &BlockArray->Entry) && (Link->ForwardLink != &BlockArray->Entry)) {
     BlockData    = BASE_CR (Link, IFR_BLOCK_DATA, Entry);
     NewBlockData = BASE_CR (Link->ForwardLink, IFR_BLOCK_DATA, Entry);
-    if ((NewBlockData->Offset >= BlockData->Offset) && ((UINTN)NewBlockData->Offset <= (UINTN)(BlockData->Offset + BlockData->Width))) {
-      if ((NewBlockData->Offset + NewBlockData->Width) > (BlockData->Offset + BlockData->Width)) {
-        BlockData->Width = (UINT16)(NewBlockData->Offset + NewBlockData->Width - BlockData->Offset);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if ((SafeUint16Add (BlockData->Offset, BlockData->Width, &Sum1) != EFI_SUCCESS) ||
+        (SafeUint16Add (NewBlockData->Offset, NewBlockData->Width, &Sum2) != EFI_SUCCESS) ||
+        (SafeUint16Sub (NewBlockData->Offset, BlockData->Offset, &Sum3) != EFI_SUCCESS))
+    {
+      continue;

```

Comment:
Here and in the other failure case we don't want to continue, right? The update step of the loop is line 2182 and if we continue here we'll end up in an infinite loop (same with 2172's continue).